qa/src/test/java/io/camunda/migrator/qa/runtime/element/SubprocessMigrationTest.java

```diff
@@ -51,4 +59,59 @@ public void migrateCallActivityAndSubprocess() {
         .hasVariable(LEGACY_ID_VAR_NAME, subProcessInstance.getProcessInstanceId());
     assertThat(byTaskName("userTaskName")).isCreated().hasElementId("userTaskId");
   }
+
+  @Test
+  public void shouldSkipMigrationWhenPropagateAllParentVariablesIsFalse() {
+    // given
+    deployer.deployProcessInC7AndC8("calledActivitySubprocess.bpmn");
+    deployer.deployProcessInC7AndC8("callActivityProcessNoPropagation.bpmn");
+    ProcessInstance parentInstance = runtimeService.startProcessInstanceByKey("callingProcessIdNoPropagation");
+
+    // when
+    runtimeMigrator.start();
+
+    // then
+    // verify no C8 instance was created
+    assertThatProcessInstanceCountIsEqualTo(0);
+
+    // verify the correct error message was logged
+    var events = logs.getEvents();
+    Assertions.assertThat(events.stream()
+        .filter(event -> event.getMessage()
+            .contains(String.format("Skipping process instance with legacyId [%s]: " + CALL_ACTIVITY_LEGACY_ID_ERROR,
+                parentInstance.getId(), "callActivityId"))))
+        .hasSize(1);
+  }
+
+  @Test
+  public void shouldNotSkipMigrationWhenPropagateAllParentVariablesIsFalseButMappingIsPresent() {
+    // given
+    deployer.deployProcessInC7AndC8("calledActivitySubprocess.bpmn");
+    deployer.deployProcessInC7AndC8("callActivityProcessNoPropagationWithMapping.bpmn");
+    ProcessInstance parentInstance = runtimeService.startProcessInstanceByKey("callingProcessIdNoPropagationWithMapping");
+    ProcessInstance subProcessInstance = runtimeService
+        .createProcessInstanceQuery()
+        .superProcessInstanceId(parentInstance.getProcessInstanceId())
+        .singleResult();
+
+    // when
+    runtimeMigrator.start();
+
+    // then
+    io.camunda.client.api.search.response.ProcessInstance c8ParentInstance =
+        camundaClient.newProcessInstanceSearchRequest().filter(processInstanceFilter -> {
+          processInstanceFilter.processDefinitionId("callingProcessIdNoPropagationWithMapping");
+        }).execute().items().getFirst();
+
+    Long c8ParentInstanceKey = c8ParentInstance.getProcessInstanceKey();
+    Optional<Variable> variable = getVariableByScope(c8ParentInstanceKey, c8ParentInstanceKey, LEGACY_ID_VAR_NAME);
+    assert variable.isPresent();
+    assert variable.get().getValue().equals("\""+parentInstance.getProcessInstanceId()+"\"");
```

Comment:
❓ For my understanding whats the difference between this explicit logic and using the `hasVariable` assertions on the callingProcessIdNoPropagationWithMapping?

```suggestion
    assertThat(byProcessId("callingProcessIdNoPropagationWithMapping")).isActive()
        .hasVariable(LEGACY_ID_VAR_NAME, parentInstance.getProcessInstanceId());
```